includes/admin/services/class-rop-facebook-service.php

```diff
@@ -458,20 +458,26 @@ public function share( $post_details, $args = array() ) {
 			return false;
 		}
 
-		if ( $this->try_post( $new_post, $args['id'], $args['access_token'] ) ) {
-			$this->logger->alert_success(
-				sprintf(
-					'Successfully shared %s to %s on %s ',
-					get_the_title( $post_details['post_id'] ),
-					$args['user'],
-					$post_details['service']
-				)
-			);
+		$rop_staging_status = new Rop();
```

Comment:
Don't instantiate the main class for the `rop_site_is_staging()` method. Instead either move the method to a utility class that can be instantiated multiple times or inject the result of the method during service factory.
I would do a `protected $is_staging` variable in the abstract service class that is instantiated in the `__construct()` with the value from the method in the utility class.